routers/api/v1/projects/project.go

```diff
@@ -0,0 +1,420 @@
+	//     in: body
+	//     required: true
+	//     schema: { "$ref": "#/definitions/UpdateProjectPayload" }
+	// responses:
+	//  "200":
+	//    "$ref": "#/responses/Project"
+	//  "403":
+	//    "$ref": "#/responses/forbidden"
+	//  "404":
+	//    "$ref": "#/responses/notFound"
+	form := web.GetForm(ctx).(*api.UpdateProjectPayload)
+	project, err := project_model.GetProjectByID(ctx, ctx.ParamsInt64("id"))
+	if err != nil {
+		if project_model.IsErrProjectNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(http.StatusInternalServerError, "UpdateProject", err)
+		}
+		return
+	}
+	if project.Title != form.Title {
+		project.Title = form.Title
+	}
+	if project.Description != form.Description {
+		project.Description = form.Description
+	}
+
+	err = project_model.UpdateProject(ctx, project)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "UpdateProject", err)
+		return
+	}
+	projectResponse, err := convert.ToAPIProject(ctx, project)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "UpdateProject", err)
+		return
+	}
+	ctx.JSON(http.StatusOK, projectResponse)
+}
+
+func DeleteProject(ctx *context.APIContext) {
+	// swagger:operation DELETE /projects/{id} project projectDeleteProject
+	// ---
+	// summary: Delete project
+	// parameters:
+	//   - name: id
+	//     in: path
+	//     description: id of the project
+	//     type: string
+	//     required: true
+	// responses:
+	//  "204":
+	//    "description": "Deleted the project"
+	//  "403":
+	//    "$ref": "#/responses/forbidden"
+	//  "404":
+	//    "$ref": "#/responses/notFound"
+
+	if err := project_model.DeleteProjectByID(ctx, ctx.ParamsInt64(":id")); err != nil {
+		ctx.Error(http.StatusInternalServerError, "DeleteProjectByID", err)
+		return
+	}
+
+	ctx.Status(http.StatusNoContent)
+}
+
+func ListUserProjects(ctx *context.APIContext) {
+	// swagger:operation GET /user/projects project projectListUserProjects
+	// ---
+	// summary: List repository projects
+	// produces:
+	// - application/json
+	// parameters:
+	//   - name: closed
+	//     in: query
+	//     description: include closed issues or not
+	//     type: boolean
+	//   - name: page
+	//     in: query
+	//     description: page number of results to return (1-based)
+	//     type: integer
+	//   - name: limit
+	//     in: query
+	//     description: page size of results
+	//     type: integer
+	// responses:
+	//  "200":
+	//    "$ref": "#/responses/ProjectList"
+	//  "403":
+	//    "$ref": "#/responses/forbidden"
+	//  "404":
+	//    "$ref": "#/responses/notFound"
+	projects, count, err := db.FindAndCount[project_model.Project](ctx, project_model.SearchOptions{
+		Type:        project_model.TypeIndividual,
+		IsClosed:    ctx.FormOptionalBool("closed"),
+		OwnerID:     ctx.Doer.ID,
+		ListOptions: db.ListOptions{Page: ctx.FormInt("page")},
+	})
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "Projects", err)
+		return
+	}
+
+	ctx.SetLinkHeader(int(count), setting.UI.IssuePagingNum)
```

Comment:
This seems wrong - why do we use UI page size inside API route, if it is provided by the parameter?